File: backend/models/Course.js
const mongoose = require('mongoose');

const reviewSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  userName: String,
  rating: {
    type: Number,
    required: true,
    min: 1,
    max: 5,
  },
  comment: String,
  date: {
    type: Date,
    default: Date.now,
  },
});

const courseSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
  },
  description: {
    type: String,
    required: true,
  },
  instructor: {
    name: String,
    title: String,
    avatar: String,
    rating: Number,
    students: Number,
    courses: Number,
  },
  category: {
    type: String,
    required: true,
  },
  tags: [String],
  level: {
    type: String,
    enum: ['Beginner', 'Intermediate', 'Advanced'],
    default: 'Beginner',
  },
  duration: {
    hours: Number,
    lectures: Number,
  },
  rating: {
    type: Number,
    default: 0,
  },
  reviewsCount: {
    type: Number,
    default: 0,
  },
  studentsEnrolled: {
    type: Number,
    default: 0,
  },
  thumbnail: String,
  videoUrl: String,
  syllabus: [{
    module: String,
    lectures: Number,
    duration: String,
    topics: [{
      title: String,
      duration: String,
      type: String,
    }],
  }],
  reviews: [reviewSchema],
  price: {
    type: Number,
    default: 0,
  },
  language: {
    type: String,
    default: 'English',
  },
  resources: {
    videos: Number,
    exercises: Number,
    downloadable: Number,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

module.exports = mongoose.model('Course', courseSchema);

File: backend-java/src/main/java/com/edulearn/backend/models/Course.java
package com.edulearn.backend.models;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Map;

@Data
@Document(collection = "courses")
public class Course {
    @Id
    private String id;
    private String title;
    private String description;
    private Instructor instructor;
    private String category;
    private List<String> tags;
    private String level;
    private Duration duration;
    private double rating;
    private int reviewsCount;
    private int studentsEnrolled;
    private String thumbnail;
    private String thumbnailColor;
    private String videoUrl;
    private List<SyllabusModule> syllabus;
    private List<Review> reviews;
    private double price;
    private String language;
    private Resources resources;
    private LocalDateTime createdAt = LocalDateTime.now();

    @Data
    public static class Instructor {
        private String name;
        private String title;
        private String avatar;
        private double rating;
        private int students;
        private int courses;
    }

    @Data
    public static class Duration {
        private Integer hours;
        private Integer lectures;
    }

    @Data
    public static class SyllabusModule {
        private String module;
        private int lectures;
        private String duration;
        private List<Topic> topics;
    }

    @Data
    public static class Topic {
        private String title;
        private String duration;
        private String type;
    }

    @Data
    public static class Review {
        private String userId;
        private String userName;
        private int rating;
        private String comment;
        private LocalDateTime date;
    }

    @Data
    public static class Resources {
        private int videos;
        private int exercises;
        private int downloadable;
    }
}
